src/libstd/collections/hash/map.rs

```diff
@@ -2445,13 +2445,18 @@ impl RandomState {
         // iteration order allows a form of DOS attack. To counter that we
         // increment one of the seeds on every RandomState creation, giving
         // every corresponding HashMap a different iteration order.
-        thread_local!(static KEYS: Cell<(u64, u64)> = {
-            Cell::new(sys::hashmap_random_keys())
+        thread_local!(static KEYS: Cell<[u64; 2]> = {
+            let mut buf = [0u8; 16];
+            // we consciously ignore potential errors here
+            let _ = getrandom::getrandom(&mut buf);
+            // SAFETY: it's safe to transmute_copy random 16 bytes to two `u64`s
+            let buf: [u64; 2] = unsafe { core::mem::transmute_copy(&buf) };
```

Comment:
To avoid unsafety you could use `u128::from_ne_bytes` and then `[n as u64, (n >> 64) as u64]`.

Reply:
Hm, it's a bit roundabout way of doing things, though indeed not a bad one. One small drawback is that on BE targets it will result in swapping bytes. I would like to hear that others think about this approach.

Reply:
On a second thought, I will use your suggestion for now.